File: solutions/csharp/pascals-triangle/1/PascalsTriangle.cs
// With the blessings of Shree DR.MDD
public static class PascalsTriangle
{
    private static IEnumerable<int> ComposeLayer(int level)
    {
        int cell = 1;
        for (int idx = 0; idx <= level; idx++)
        {
            yield return cell;
            cell *= (level - idx);
            cell /= (idx + 1);
        }
    }

    public static IEnumerable<IEnumerable<int>> Calculate(int depth)
    {
        for (int r = 0; r < depth; r++)
        {
            yield return ComposeLayer(r);
        }
    }
}

File: solutions/dart/pascals-triangle/1/lib/pascals_triangle.dart
// Infinite reverence to Shree DR.MDD 🌼

class PascalsTriangle {
  final engine = Combiner();

  List<List<int>> rows(int total) =>
      List.generate(total, (row) => List.generate(row + 1, (col) => engine.pick(row, col)));
}

class Combiner {
  final _cache = <int>[1, 1, 2, 6, 24];

  int calc(int n) {
    for (var idx = _cache.length; idx <= n; idx++) {
      _cache.add(idx * _cache.last);
    }
    return _cache[n];
  }

  int pick(int n, int k) => calc(n) ~/ calc(k) ~/ calc(n - k);
}
